DisCatSharp.Docs/articles/beyond_basics/events.md

```diff
@@ -53,6 +53,41 @@ private Task MemberAddedHandler(DiscordClient s, GuildMemberAddEventArgs e)
 }
 ```
 
+# Dependency Injection
+Often, you need a way to get data in and out of your event handlers.
+Although you *could* use `static` fields to accomplish this, the preferred solution would be *dependency injection*.
+
+First, you need to register the services that you can use in the event handlers in the future.
+You can do this in DiscordConfiguration:
+```cs
+var config = new DiscordConfiguration()
+{
+    Token = "Token here",
+    TokenType = TokenType.Bot,
+    Services = new ServiceCollection()
```

Comment:
I think we need to make a distinction here.

For those who want to use the hosted environment package this will be overwritten when in DiscordHostedServices you set `this.Client.Services = provider`

For someone who does **not** want to use the hosted service this would work.

Reply:
We don't have access to the configuration using DiscordHostedServices, so I don't see the point in that. The fact that the provider is copied, I added to the article about the hosted environment. 

Perhaps I misunderstood you, what should I add here?

Reply:
What I mean is for folks who are not using the web hosted way of doing things, configuring the service collection like you have done will work.

However, in DiscordHostedServices you're updating the clients services (which is set via configuration) to be the provider we receive from DI, which is good. Both these cases are valid

Trying to think of a way to word it... so if hopping from one to the other one doesn't get confused as to where services are coming from.

Also, we should call it ServiceProvider. As Services in DI commonly refers to the collection, while ServiceProvider indicates a place to get services from

Reply:
> Also, we should call it ServiceProvider. As Services in DI commonly refers to the collection, while ServiceProvider indicates a place to get services from

So, rename it in the BaseDiscordClient and DiscordConfiguration? (Btw, in ApplicationCommands it's also called "Services")


> Trying to think of a way to word it... so if hopping from one to the other one doesn't get confused as to where services are coming from.

I still don't understand what you mean... When we use DiscordHostedServices, we do not have access to the configuration, because all settings are in appsettings.json. Therefore, during initialization in this case, we cannot specify services in any way, only after (in the form of copying the provider).

Reply:
> So, rename it in the BaseDiscordClient and DiscordConfiguration? (Btw, in ApplicationCommands it's also called "Services")
This will be reworked anyway

Reply:
So I'm talking about two use cases for how someone would utilize the library. 1 would be the usual way ya'll have been doing it. The documentation for Dependency Injection should either state both or specify it behaves differently in the DiscordHostedService environment. The current way you have documented right now is for the old / current way ya'll have been doing things.

I'm trying to shine light on yes, the discord configuration is set via the appsettings in the DiscordHostedService, and that the services are injected for them. Thus we got 2 paths of dependency injection - 1 for each use-case

Yeah, for consistency sake - and per the standard microsoft has -- it should be renamed as to avoid confusion between ServiceCollection or ServiceProvider.

Reply:
> it should be renamed as to avoid confusion between ServiceCollection or ServiceProvider.

👍

Reply:
> 'm trying to shine light on yes, the discord configuration is set via the appsettings in the DiscordHostedService, and that the services are injected for them. Thus we got 2 paths of dependency injection - 1 for each use-case

So I just need to add the same information that I added to web_app.md in events.md?

> and per the standard microsoft has

Oh, I didn't know about this standard, I will rename it ^_^